File: Bank.java
import java.util.ArrayList;

public class Bank {
    private ArrayList<BankAccount> cuentas;

    public Bank(){
        this.cuentas = new ArrayList<>();
    }

    public void agregarCuenta(BankAccount cuenta){
        this.cuentas.add(cuenta);
    }

    public void listarCuentas(){
        if (cuentas.isEmpty()){
            System.out.println("ERROR. No se ha agregado ninguna cuenta.");
            System.out.println("");
        }
        else{
            System.out.println("=== CUENTAS AGREGADAS ===");
            for(BankAccount cuenta : cuentas){
                System.out.println("*******************************************");
                System.out.println(String.format("Propietario de la cuenta: %s %s", cuenta.getOwnerName(), cuenta.getOwnerLastName()));
                System.out.println("Número de cuenta: " + cuenta.getAccountNumber());
                System.out.println("Tipo de cuenta: " + cuenta.getType());
                System.out.println("*******************************************\n");
            }
        }
    }

    public void mostrarCuentasUsuario(String ownerName, String ownerLastName) {
        boolean encontrado = false;
        for (BankAccount cuenta : cuentas) {
            if (cuenta.getOwnerName().equals(ownerName) && cuenta.getOwnerLastName().equals(ownerLastName)) {
                encontrado = true;
                System.out.println("*******************************************");
                System.out.println(String.format("Propietario de la cuenta: %s %s", cuenta.getOwnerName(), cuenta.getOwnerLastName()));
                System.out.println("Número de cuenta: " + cuenta.getAccountNumber());
                System.out.println("Tipo de cuenta: " + cuenta.getType());
                System.out.println("*******************************************\n");
            }
        }
        if (!encontrado) {
            System.out.println(String.format("No se encontraron cuentas para el usuario %s %s", ownerName, ownerLastName));
            System.out.println();
        }
    }

    public void mostrarCuenta(long accountNumber){
        boolean encontrado = false;
        for (BankAccount cuenta : cuentas){
            if (cuenta.getAccountNumber() == accountNumber){
                encontrado = true;
                System.out.println("*******************************************");
                System.out.println(String.format("Propietario de la cuenta: %s %s", cuenta.getOwnerName(), cuenta.getOwnerLastName()));
                System.out.println("Número de cuenta: " + cuenta.getAccountNumber());
                System.out.println("Tipo de cuenta: " + cuenta.getType());
                System.out.println("*******************************************\n");
            }
        }
        if (!encontrado){
            System.out.println("No se encontraron cuentas para el número " + accountNumber);
            System.out.println();
        }
    }

}

File: bank.py

class Bank:
    __accounts = []

    def __init__(self):
        self.__accounts = []

    def add_account(self, account):
        self.__accounts.append(account)

    def list_accounts(self):
        print("=== CUENTAS AGREGADAS ===")
        for account in self.__accounts:
            print("*******************************************")
            print("Propietario de la cuenta: {} {}".format(account.get_owner_name(), account.get_owner_last_name()))
            print("Número de cuenta: {}".format(account.get_account_number()))
            print("Tipo de cuenta: {}".format(account.get_type()))
            print("*******************************************\n")

    def show_user_account(self, owner_name, owner_last_name):
        encontrado = False
        for account in self.__accounts:
            if account.get_owner_name() == owner_name:
                encontrado = True
                print("*******************************************")
                print("Propietario de la cuenta: {} {}".format(account.get_owner_name(), account.get_owner_last_name()))
                print("Número de cuenta: {}".format(account.get_account_number()))
                print("Tipo de cuenta: {}".format(account.get_type()))
                print("*******************************************\n")
        if not encontrado:
            print("No se encontraron cuentas para el usuario {}. {}.".format(owner_name, owner_last_name))

    def show_account(self, account_number):
        encontrado = False
        for account in self.__accounts:
            if account.get_account_number() == account_number:
                encontrado = True
                print("*******************************************")
                print("Propietario de la cuenta: {} {}".format(account.get_owner_name(), account.get_owner_last_name()))
                print("Número de cuenta: {}".format(account.get_account_number()))
                print("Tipo de cuenta: {}".format(account.get_type()))
                print("*******************************************\n")
        if not encontrado:
            print("No se encontraron cuentas para el número {}.".format(account_number))
